tests/system/client.go

```diff
@@ -72,25 +77,41 @@ func waitForSession(akPath, akAddr, step string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("invalid duration %q: %w", match[1], err)
 	}
-	id := match[2]
+	isSessionExpected := true // wait .. for .. session
+	if match[2] == "unless" {
+		isSessionExpected = false // wait .. unless .. session
+	}
+	waitType := match[2]
+	id := match[3]
 
 	// Check the session state with the AK client.
-	state := regexp.MustCompile(`state:SESSION_STATE_TYPE_(COMPLETED|ERROR|STOPPED)`)
+	stateRegex := sessionStateFinal // wait for session to finish
+	if !isSessionExpected {
+		stateRegex = sessionStateAll // any state would mean error
+	}
+
 	args := append(config.ServiceUrlArg(akAddr), "session", "get", id)
 	startTime := time.Now()
 
+	sessionFound := false
 	for time.Since(startTime) < duration {
 		result, err := runClient(akPath, args)
 		if err != nil {
 			return "", fmt.Errorf("failed to get session: %w", err)
 		}
-		if state.MatchString(result.output) {
+		if stateRegex.MatchString(result.output) {
```

Comment:
No need for `sessionFound = true` below.

```suggestion
		sessionFound = stateRegex.MatchString(result.output)
		if sessionFound {
```

Reply:
done